src/main/kotlin/no/nav/helse/flex/oppdatersporsmal/muteringer/ArbeidGjenopptattMutering.kt

```diff
@@ -97,3 +100,67 @@ fun Sykepengesoknad.arbeidGjenopptattMutering(): Sykepengesoknad {
 
     return this.leggTilSporsmaal(oppdaterteSporsmal)
 }
+
+private fun Sykepengesoknad.arbeidGjenopptattMuteringSelvstendigFrilanser(): Sykepengesoknad {
+    val arbeidGjenopptattDato = this.finnGyldigDatoSvar(TILBAKE_I_ARBEID, TILBAKE_NAR)
+    if (arbeidGjenopptattDato != null) {
+        if (arbeidGjenopptattDato == this.fom) {
+            // Fjerner spørsmål som forsvinner
+            return this.copy(
+                sporsmal = sporsmal
+                    .asSequence()
+                    .filterNot { (_, tag) -> tag.startsWith(JOBBET_DU_GRADERT) }
+                    .filterNot { (_, tag) -> tag.startsWith(JOBBET_DU_100_PROSENT) }
+                    .filterNot { (_, tag) -> tag.startsWith(ARBEID_UNDERVEIS_100_PROSENT) }
+                    .filterNot { (_, tag) -> tag == FERIE_PERMISJON_UTLAND }
+                    .filterNot { (_, tag) -> tag == FERIE_V2 }
+                    .filterNot { (_, tag) -> tag == PERMISJON_V2 }
+                    .filterNot { (_, tag) -> tag == UTLAND_V2 }
+                    .filterNot { (_, tag) -> tag == UTLAND }
+                    .filterNot { (_, tag) -> tag == UTDANNING }
+                    .toMutableList()
+            )
+        }
+    }
+
+    val oppdatertTom = if (arbeidGjenopptattDato == null) {
+        this.tom
+    } else {
+        arbeidGjenopptattDato.minusDays(1)
+    }
+
+    val oppdaterteSporsmal = jobbetDuIPeriodenSporsmalSelvstendigFrilanser(
+        this.skapOppdaterteSoknadsperioder(
+            arbeidGjenopptattDato
+        ),
+        this.arbeidssituasjon!!,
+    ).toMutableList()
+
+    if (!oppdatertTom!!.isBefore(this.fom)) {
+        if (harFeriePermisjonEllerUtenlandsoppholdSporsmal()) {
+            oppdaterteSporsmal.add(
+                gammeltFormatFeriePermisjonUtlandsoppholdSporsmal(
+                    this.fom!!,
+                    oppdatertTom
+                )
+            )
+        } else {
+            if (this.soknadstype == ARBEIDSTAKERE) {
```

Comment:
Denne burde vel sjekkes i toppen og enten kaste feil eller bare returnere

Reply:
nja, den sjekkes i starten. Men problemet er at det er forskjellige spørsmål som legges til for frilanser og arbeidstaker. Det er det denne iffen sjekker. Men ser jeg rota litt til i koden, kommer en ny push